File: LeetCode/2293. Min Max Game/solution.cpp
/**
 * Runtime: 0 ms
 * Memory Usage: 9.7 MB 
 */
class Solution {
public:
    int minMaxGame(vector<int>& nums) {
        int n = nums.size();
        if (n == 1) {
            return nums[0];
        }
        vector<int> newNums(n >> 1);
        for (int i = 0; i < newNums.size(); ++ i) {
            if (i & 1) {
                newNums[i] = max(nums[2 * i], nums[2 * i + 1]);
            } else {
                newNums[i] = min(nums[2 * i], nums[2 * i + 1]);
            }
        }
        return minMaxGame(newNums);
    }
};

File: LeetCode/2293. Min Max Game/solution.py
"""
Runtime: 40 ms
Memory Usage: 15 MB
"""
class Solution:
    def minMaxGame(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 1:
            return nums[0]
        new_nums = [0] * (n >> 1)
        for i in range(n >> 1):
            if i % 2 == 0:
                new_nums[i] = min(nums[i * 2], nums[i * 2 + 1])
            else:
                new_nums[i] = max(nums[i * 2], nums[i * 2 + 1])
        return self.minMaxGame(new_nums)
